{{cookiecutter.project_slug}}/.github/workflows/test-pr.yml

```diff
@@ -0,0 +1,85 @@
+name: 'Run Tests'
+on:
+  pull_request:
+  workflow_dispatch:
+concurrency:
+  group: {% raw %}${{ github.workflow }}-${{ github.ref }}{% endraw %}
+  cancel-in-progress: true
+
+jobs:
+  code-quality-checks:
+    name: 'Code Quality Checks'
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Check out code'
+        uses: actions/checkout@v3
+      - name: 'Install dependencies'
+        run: |
```

Comment:
We should put a big warning here: "IF YOU ARE ON A PRIVATE REPO, SWITCH THIS TO PRIVATE RUNNER", otherwise when people copy this template they will very quickly wrack up github bills.

Reply:
I feel adding these CI scripts to the template is not the right solution. Python and CI are two separate concerns, and we should handle them as such.

So instead, we could ether
a) Create a repo with well-documented scripts, actions, etc. that downstream projects can import.
b) Create a repo with a CI template. We can include more complex logic to control what sort of scripts are generated (volume vs cp, public vs private, Docker vs host, etc.).

Then the Python template can still be used without committing to the way we do CI in our infrastructure.

Reply:
Hmmmmm, you have a good point there. Maybe this belongs in devops or devops-actions repo instead. For now, leave this as a draft until I can discuss more with @F-WRunTime .